### ISSUE TITLE
Check for invalid target_arch

### ISSUE THREAD

**Issue by asomers** [rust-lang/rust-clippy#7828]
### What it does

Checks if the value used with `#[cfg(target_arch = "XXX")]` is valid.

### Categories (optional)

- Kind: correctness

For example:
- Detects dead code that almost certainly shouldn't be dead

### Drawbacks

Might not be forwards-compatible with architectures added by future compilers.  For example, if Rust 1.75.0 adds support for itanium and a crate cfg-gates on that, Clippy 1.74.0 would falsely warn about it.

### Example

A real-world example from the Nix crate: https://github.com/nix-rust/nix/pull/1566
```rust
#[cfg(target_arch = "ppc")]
pub struct Foo{}
```

Should be written as:

```rust
#[cfg(any(target_arch = "powerpc", target_arch = "powerpc64"))]
pub struct Foo{}
```

**Comment by asomers** [comment]
Also, checking for invalid `target_env` would be good.  I just discovered some code that was wrongly checking for `target_env = "newlibc"` instead of `target_env = "newlib"`.

**Comment by xFrednet** [comment]
Hey, thank you for the suggestion. In general, it could be nice to check for valid targets etc.

The current API of rustc(which Clippy uses) sadly doesn't really support lints that check code before the expansion of macros. At one point, we want to have some kind of interface to also lint macros, like in this example. For now, I'll add the *blocked* label until we find a good way to handle macros :upside_down_face:
